pkg/render/apiserver.go

```diff
@@ -2325,6 +2327,16 @@ func (c *apiServerComponent) getDeprecatedResources() []client.Object {
 				Name: "tigera-tier-getter",
 			},
 		})
+
+		// Delete the older namespace for enterprise
+		if c.cfg.CanCleanupOlderResources {
```

Comment:
I must be missing something, but where is the deployment cleaned up?

Reply:
I thought deleting a namespace does trigger garbage collection for all namespaced resources within it.
API Server switching from OSS to Enterprise https://github.com/tigera/operator/blob/b72f4372a9fa1855fc1caa9c0400ed2374675192/pkg/render/apiserver.go#L246
Moving Guardian from `tigera-guardian` to `calico-system`
https://github.com/tigera/operator/blob/b72f4372a9fa1855fc1caa9c0400ed2374675192/pkg/render/guardian.go#L695

I observed the same behavior during my testing, the namespace enters `Terminating` and eventually all resources are cleaned up as expected. I’ve also included this behavior as part of the test coverage.

WDYT? Do we need to explicitly delete each resource inside `tigera-system` before removing the namespace?